Implement `macaddr8` type

benoit9126 opened the pull request:
Hi,

I am trying to implement `macaddr8` PostgreSQL type as defined [here](https://www.postgresql.org/docs/10/datatype-net-types.html#DATATYPE-MACADDR8). Here is a proposal.

jimfunk commented:
Oh this is great!

Unfortunately, some tests are not passing:

```
======================================================================
ERROR: test_retrieves_eui_type (test.tests.test_sql_fields.TestMACAddress8FieldArray)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jfunk/projects/django-postgresql-netfields/test/tests/test_sql_fields.py", line 743, in test_retrieves_eui_type
    instance = MACArrayTestModel.objects.get(id=instance.id)
  File "/home/jfunk/projects/django-postgresql-netfields/.tox/py38-django31/lib/python3.8/site-packages/django/db/models/manager.py", line 85, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
  File "/home/jfunk/projects/django-postgresql-netfields/.tox/py38-django31/lib/python3.8/site-packages/django/db/models/query.py", line 429, in get
    raise self.model.DoesNotExist(
test.models.MACArrayTestModel.DoesNotExist: MACArrayTestModel matching query does not exist.

======================================================================
FAIL: test_24bit_colon (test.tests.test_form_fields.TestMacAddress8FormField)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jfunk/projects/django-postgresql-netfields/test/tests/test_form_fields.py", line 275, in test_24bit_colon
    self.assertTrue(form.is_valid())
AssertionError: False is not true

======================================================================
FAIL: test_24bit_hyphen (test.tests.test_form_fields.TestMacAddress8FormField)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jfunk/projects/django-postgresql-netfields/test/tests/test_form_fields.py", line 280, in test_24bit_hyphen
    self.assertTrue(form.is_valid())
AssertionError: False is not true

======================================================================
FAIL: test_32bit_colon (test.tests.test_form_fields.TestMacAddress8FormField)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jfunk/projects/django-postgresql-netfields/test/tests/test_form_fields.py", line 285, in test_32bit_colon
    self.assertTrue(form.is_valid())
AssertionError: False is not true

======================================================================
FAIL: test_32bit_hyphen (test.tests.test_form_fields.TestMacAddress8FormField)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jfunk/projects/django-postgresql-netfields/test/tests/test_form_fields.py", line 290, in test_32bit_hyphen
    self.assertTrue(form.is_valid())
AssertionError: False is not true

======================================================================
FAIL: test_trunc (test.tests.test_functions.TestMac8FieldFunctions)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/jfunk/projects/django-postgresql-netfields/test/tests/test_functions.py", line 268, in test_trunc
    self.assertEqual(qs[0].trunc, EUI('88:99:aa:bb:cc:00:00:00'))
AssertionError: EUI('88:99:aa:00:00:00:00:00') != EUI('88-99-AA-BB-CC-00-00-00')

----------------------------------------------------------------------
```

This reminds me that I need to figure out why Travis broke for PRs. :-)

benoit9126 commented:
@jimfunk Thanks. I hoped to find the link to Travis CI to see tests results...

- I overestimated the ability of `EUI` object to parse Mac addresses in the previous tests. I corrected them.
- I added the function `macaddr8_set7bit` + tests
- I also added some extra tests in tox.ini files

jimfunk commented:
Thanks!
